qtt/algorithms/functions.py

```diff
@@ -233,7 +233,7 @@ def exp_function(x, a, b, c):
     return y
 
 
-def cost_exp_decay(x_data, y_data, params, threshold=None):
+def cost_exp_decay(x_data, y_data, params, threshold=None, offset_parameter = None):
```

Comment:
There is an argument `params` which can set the `a` argument of `exp_function`. Therefore I do not think the `offset_parameter` argument is needed.

I think you can do use the `params` in a more fancy way, by making use of `**kwargs`.

Reply:
Fair point. I came up with a solution. To test the speed I used the following:
```
import numpy.core.numeric
import numpy as np
from qtt.algorithms.functions import *

params=[2., 3.]
offset_parameter=2.2

x_data=np.random.rand()
y_data=exp_function(x_data, offset_parameter, *params)+np.random.rand()

threshold=1.

#%%

def func1(params): return cost_exp_decay(x_data, y_data, params, threshold, None)

print(func1([offset_parameter,1,1]))

# %timeit func1([offset_parameter, 1,1]): 8,5 us

#%% Current method
def func2(params): return cost_exp_decay(x_data, y_data, params, threshold, offset_parameter)

print(func2([1,1]) )

# %timeit func2([1,1]): 10.2 us 

#%% Alternative method (would simplify cost_exp_decay since offset_parameter is not needed any more)



def func3(params): return cost_exp_decay(x_data, y_data, np.hstack( (offset_parameter, params)), threshold)

# not much faster
#def func3(params): return cost_exp_decay(x_data, y_data, numpy.core.numeric.concatenate(([offset_parameter], np.array(params) ), 0), threshold)
# fast but not robust for all input types
#def func3(params): return cost_exp_decay(x_data, y_data, [offset_parameter]+params, threshold)

print(func3([1,1]) )
```
The third options does what we want, but it quite a bit. Do you see alternatives?

@CumulonimbusCalvus